x/mongo/driver/topology/server_test.go

```diff
@@ -274,86 +239,6 @@ func TestServerConnectionTimeout(t *testing.T) {
 	}
 }
 
```

Comment:
IIUC the tests below, are no longer as relevant since the tricky logic of `ProcessHandshakeError` added in GODRIVER-2037 has been reverted.